.github/workflows/build-and-test.yml

```
@@ -0,0 +1,305 @@
+    - uses: actions/checkout@v3
+    - uses: pnpm/action-setup@v4
+      with:
+        package_json_file: tests/suite/package.json
+    - uses: dtolnay/rust-toolchain@stable
+    - uses: Swatinem/rust-cache@v2
+      with:
+        # The prefix cache key, this can be changed to start a new cache manually.
+        # default: "v0-rust"
+        prefix-key: v0
+        # Cache only the cargo registry
+        cache-targets: false
+    - uses: mozilla-actions/sccache-action@v0.0.3
+    - name: Check fmt
+      run: make -C tests check-fmt
+    - name: Check clippy
+      run: make -C tests check-clippy
+
+
+  generate-matrix:
+    name: Generate Integration Test Matrix
+    runs-on: ubuntu-latest
+    outputs:
+      networks: ${{ steps.generate-matrix.outputs.networks }}
+      build_tag: ${{ steps.generate-matrix.outputs.build_tag }}
+    steps:
+      -
+        name: Checkout
+        uses: actions/checkout@v3
+      -
+        name: Generate network matrix
+        id: generate-matrix
+        run: |
+          NETWORKS=$(ls tests/networks | jq -R -s -c '. | gsub(".yaml"; "") | split("\n")[:-1]')
+          echo "Networks:"
+          echo ${NETWORKS}
+          echo "networks=${NETWORKS}" >> $GITHUB_OUTPUT
+          # Choose unique name for this build
+          BUILD_TAG="$(echo ${{ github.sha }} | head -c 8)-${{ github.run_id }}"
+          echo "Build tag:"
+          echo ${BUILD_TAG}
+          echo "build_tag=${BUILD_TAG}" >> $GITHUB_OUTPUT
+          cat $GITHUB_OUTPUT
+
+  publish-suite:
+    name: Publish Integration Test Suite
+    runs-on: ubuntu-latest
+    needs: generate-matrix
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v2
+    - name: Login to Public ECR
+      uses: docker/login-action@v2
+      with:
+        registry: public.ecr.aws
+        username: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        password: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+      env:
+        AWS_REGION: us-east-1
+    - name: Build and Publish
+      run: make -C tests BUILD_TAG=${{ needs.generate-matrix.outputs.build_tag }} publish-suite
+
+
+  build-driver:
+    name: Builder hermetic-driver
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: dtolnay/rust-toolchain@stable
+    - uses: Swatinem/rust-cache@v2
+      with:
+        # The prefix cache key, this can be changed to start a new cache manually.
+        # default: "v0-rust"
+        prefix-key: v0
+        # Cache only the cargo registry
+        cache-targets: false
+    - uses: mozilla-actions/sccache-action@v0.0.3
+    - name: Build Tester
+      run: make -C tests BUILD_PROFILE=release driver
+    - uses: actions/upload-artifact@master
+      with:
+        name: hermetic-driver
+        path: ./tests/target/release/hermetic-driver
+        retention-days: 1
+
+  run-integration-tests:
+    name: Run Integration Tests
+    runs-on: ubuntu-latest
+    environment: dev-qa-2024
```

Can move this to tnet/prod cluster now.
```suggestion
    environment: tnet-prod-2024
```